src/goto-instrument/nondet_volatile.h

```diff
@@ -14,6 +14,41 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include <goto-programs/goto_model.h>
 
-void nondet_volatile(goto_modelt &);
+class cmdlinet;
+class optionst;
+
+// clang-format off
+#define NONDET_VOLATILE_OPT "nondet-volatile"
+#define NONDET_VOLATILE_VARIABLE_OPT "nondet-volatile-variable"
+
+#define OPT_NONDET_VOLATILE \
+  "(" NONDET_VOLATILE_OPT ")" \
+  "(" NONDET_VOLATILE_VARIABLE_OPT "):"
+
+#define HELP_NONDET_VOLATILE \
+  " --" NONDET_VOLATILE_OPT "            makes reads from volatile variables " \
+  "non-deterministic\n" \
+  " --" NONDET_VOLATILE_VARIABLE_OPT " <variable>\n" \
+  "                              makes reads from given volatile variable " \
+  "non-deterministic\n"
```

Comment:
:bulb: Not sure if even possible with the way parse options work, but I think a nicer syntax from a user perspective might be:

--nondet-volatile -> all volatile readss nondet
 --nondet-volatile x -> reads from x nondet 

Rather than having separate flags

Reply:
Currently not possible as the command line parsing would need to disambiguate whether the thing following `--nondet-volatile` is an argument to it, or an argument to goto-instrument or another option.